src/librustc_data_structures/stable_map.rs

```diff
@@ -0,0 +1,80 @@
+pub use rustc_hash::FxHashMap;
+use std::borrow::Borrow;
+use std::collections::hash_map::{Entry, IterMut};
+use std::hash::Hash;
+
+/// A deterministic wrapper around FxHashMap that does not provide iteration support.
+///
+/// It supports insert, remove, get and get_mut functions from FxHashMap.
+/// It also allows to convert hashmap to a sorted vector with the function `into_sorted_vector()`.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct StableMap<K, V>
+where
+    K: Eq + Hash,
+{
+    base: FxHashMap<K, V>,
+}
+
+impl<K: Hash + Eq + Ord, V: Ord> StableMap<K, V> {
+    pub fn new() -> StableMap<K, V> {
+        StableMap { base: FxHashMap::default() }
+    }
+
+    pub fn into_sorted_vector(self) -> Vec<(K, V)>
+    where
+        V: PartialOrd,
+    {
+        let mut v = self.base.into_iter().collect::<Vec<_>>();
+        v.sort_unstable();
+        v
+    }
+}
+
+impl<K, V> StableMap<K, V>
+where
+    K: Eq + Hash,
+{
+    pub fn entry(&mut self, k: K) -> Entry<'_, K, V> {
+        self.base.entry(k)
+    }
+
+    pub fn get<Q: ?Sized>(&self, k: &Q) -> Option<&V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.get(k)
+    }
+
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.get_mut(k)
+    }
+
+    pub fn insert(&mut self, k: K, v: V) -> Option<V> {
+        self.base.insert(k, v)
+    }
+
+    pub fn remove<Q: ?Sized>(&mut self, k: &Q) -> Option<V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.remove(k)
+    }
+}
+
+impl<'a, K, V> IntoIterator for &'a mut StableMap<K, V>
```

Comment:
Please remove this. `HashMap` iteration order is non deterministic, while `StableMap` should be deterministic.

Reply:
This is embarrassing. I was probably trying this out and forgot to remove it. Sorry about this.